src/algorithms/groundstate/dmrg.jl

```diff
@@ -20,30 +20,25 @@ end
 
         for pos=1:(length(state)-1)
             (eigvals,vecs) =  let state = state,parameters = parameters
```

Comment:
This is a completely independent remark: I don't think the `let` construction has much effect on type stability here. If I recall correctly, when using arguments of the function (here `state` and `parameters`) in closures, their type is fixed and therefore the `let` block is not necessary. If there would be a reason to use a `let` block, it would be for the variable `pos` which is used in the closure. Ideally, of course, no `let` block is needed altogether.

Reply:
I recall this being needed on an older version of julia - I haven't really tested it recently. I'll check it again because if we can get rid of those let blocks, the code would look much cleaner :)